src/Agda2Hs/Compile/Type.hs

```diff
@@ -184,14 +186,21 @@ compileTeleBinds tel =
 compileKeptTeleBind :: Hs.Name () -> Type -> C (Hs.TyVarBind ())
 compileKeptTeleBind x t = do
   checkValidTyVarName x
-  case compileKind t of
+  compileKind t >>= (\case
     Just k              -> pure $ Hs.UnkindedVar () x -- In the future we may want to show kind annotations
     _                   -> genericDocError =<<
       text "Kind of bound argument not supported:"
-      <+> parens (text (Hs.prettyPrint x) <> text " : " <> prettyTCM t)
+      <+> parens (text (Hs.prettyPrint x) <> text " : " <> prettyTCM t))
 
-compileKind :: Type -> Maybe (Hs.Kind ())
+compileKind :: Type -> C (Maybe (Hs.Kind ()))
```

Comment:
Actually I am stupid: `compileKind` was not a monadic function so there's no context to worry about. So using `underAbstraction` here is not really necessary after all. I'm sorry for the extra work, but could you make it pure again?

Reply:
Thanks!